src/lrx_processor.cc

```diff
@@ -272,541 +254,29 @@ LRXProcessor::recognisePattern(const wstring lu, const wstring op)
-    }
-
-
-    // We're reading a tag
-    if(val == L'<' && !isEscaped && !outOfWord)
-    {
-      wstring tag = L"";
-      tag = readFullBlock(input, L'<', L'>');
-      sl[pos] = sl[pos] + tag;
-      val = static_cast<int>(alphabet(tag));
-      if(val == 0)
-      {
-        val = static_cast<int>(alphabet(L"<ANY_TAG>"));
-      }
-      if(debugMode)
-      {
-        fwprintf(stderr, L"tag %S: %d\n", tag.c_str(), val);
-      }
-    }
-
-    if(!outOfWord)
-    {
-      if(debugMode)
-      {
-        fwprintf(stderr, L"outOfWord = false\n");
-      }
-
-      new_states.clear();
-      wstring res = L"";
-      for(vector<State>::const_iterator it = alive_states.begin(); it != alive_states.end(); it++)
-      {
-        res = L"";
-        State s = *it;
-        if(val < 0)
-        {
-          alphabet.getSymbol(res, val,  false);
-          if(debugMode)
-          {
-            fwprintf(stderr, L"  step: %S\n", res.c_str());
-          }
-          s.step(val, alphabet(L"<ANY_TAG>"));
-        }
-        else
-        {
-          if(debugMode)
-          {
-            fwprintf(stderr, L"  step: %C\n", val);
-          }
-          s.step_case(val, alphabet(L"<ANY_CHAR>"), false);
-        }
-        if(s.size() > 0) // If the current state has outgoing transitions, add it to the new alive states
-        {
-          new_states.push_back(s);
-        }
-      }
-      if(debugMode)
-      {
-        fwprintf(stderr, L"new_states: %d\n", new_states.size());
-      }
-      alive_states.swap(new_states);
-      alive_states.push_back(*initial_state);
-
-    }
-
-    // We're still reading a surface form
-    if(val > 0 && val != L'$' && !isEscaped && !outOfWord)
-    {
-      sl[pos] = sl[pos] + static_cast<wchar_t>(val);
-    }
-
-    // Reading a superblank
-    if(outOfWord)
-    {
-      if(!feof(input))
-      {
-        blanks[pos] = blanks[pos] + static_cast<wchar_t>(val);
-      }
-      if(debugMode)
-      {
-        //fwprintf(stderr, L"blanks[%d] = %S\n", pos, blanks[pos].c_str());
-      }
-    }
-
-    // Increment the current line number (for rule tracing)
-    if(val == L'\n')
-    {
-      lineno++;
-    }
-  }
-
-  processFlush(output, sl, tl, blanks, covers, empty_seq, spans, last_final);
-
-  fwprintf(output, L"%S", blanks[pos].c_str());
-}
-*/
-
-void
-LRXProcessor::process(FILE *input, FILE *output)
-{
-  bool isEscaped = false;
-
-  map<int, wstring > sl; // map of SL words
-  map<int, vector<wstring> > tl; // map of vectors of TL translations
-  map<int, wstring > blanks; // map of the superblanks
-
-  map<int, map<wstring, double> > scores; //
-  map<int, map<wstring, OpType> > operations;
+  map<int, map<UString, double> > scores; //
+  map<int, map<UString, OpType> > operations;
 
   vector<State*> alive_states ;
   alive_states.push_back(new State(*initial_state));
 
-  int val = 0;
-  while((val = fgetwc_unlocked(input)) != EOF && val != WEOF)
+  int32_t val = 0;
+  while((val = input.get()) != U_EOF)
```

Comment:
Iirc, other codebases still used `EOF` in places. Should double-check semantics of all `EOF`.

Reply:
`grep` doesn't find me any places that mix `EOF` with `InputFile` or `UFILE`.